src/Bindings/LuaState.h

```diff
@@ -112,6 +115,63 @@ class cLuaState
 	} ;
 	
 	
+	/** Represents a callback to Lua that C++ code can call.
+	Is thread-safe and unload-safe.
+	When the Lua state is unloaded, the callback returns an error instead of calling into non-existent code.
+	To receive the callback instance from the Lua side, use RefStack() or (better) cLuaState::GetStackValue(). */
+	class cCallback
+	{
+	public:
+		/** Creates an unbound callback instance. */
+		cCallback(void) = default;
+
+		~cCallback()
+		{
+			Clear();
+		}
+
+		/** Calls the Lua callback, if still available.
+		Returns true if callback has been called.
+		Returns false if the Lua state isn't valid anymore. */
+		template <typename... Args>
+		bool Call(Args &&... args)
```

Comment:
Two reasons:
- I don't want to run into the same problem as `cWorld::ScheduleTask()`, which now accepts a `std::function` and so can accept lambdas, but if you pass in a lambda, the server will crash because the lambda object is no longer valid when the scheduled callback is called
- I'm using a `SharedPtr<cCallbackContainer>` for storage, so actually caling the callback is via `container->m_Callback->Call(...)`; if I had `operator ()`, the call would look weird: `container->m_Callback->operator()(...)`